From 650e8069d41b6be4b0b9250731ed70c7f54c1801 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:47 -0400
Subject: [ethernet] i40e: limit GLLAN_TXPRE_QDIS to QINDX 0-127

Message-id: <1409658042-22689-227-git-send-email-sassmann@redhat.com>
Patchwork-id: 90415
O-Subject: [RHEL7.1 PATCH 226/281] i40e: limit GLLAN_TXPRE_QDIS to QINDX 0-127
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Christopher Pau <christopher.pau@intel.com>

Prevent writing to reserved bits, queue index is 0-127

Change-ID: Ic923e1c92012a265983414acd8f547c4bdac2e34
Signed-off-by: Christopher Pau <christopher.pau@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 24a768cfc4f8cbbb3eaf56284610e652ef05fc5b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_common.c b/drivers/net/ethernet/intel/i40e/i40e_common.c
index 8305c8a..9d09ab3 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_common.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_common.c
@@ -669,8 +669,10 @@ void i40e_pre_tx_queue_cfg(struct i40e_hw *hw, u32 queue, bool enable)
  u32 reg_block = 0;
  u32 reg_val;
 
- if (abs_queue_idx >= 128)
+ if (abs_queue_idx >= 128) {
   reg_block = abs_queue_idx / 128;
+  abs_queue_idx %= 128;
+ }
 
  reg_val = rd32(hw, I40E_GLLAN_TXPRE_QDIS(reg_block));
  reg_val &= ~I40E_GLLAN_TXPRE_QDIS_QINDX_MASK;
-- 
1.7.1